pkg/localstore/subscription_push.go

```diff
@@ -73,12 +75,28 @@ func (db *DB) SubscribePush(ctx context.Context) (c <-chan swarm.Chunk, stop fun
 						return true, err
 					}
 
+					// check if we reached item that was already processed
+					// and stop
+					if toItemKey != nil {
+						dataItemKey, err := db.pushIndex.ItemKey(dataItem)
+						if err != nil {
+							return true, err
+						}
+
+						if bytes.Equal(dataItemKey, toItemKey) {
+							toItemKey = nil
+							return true, nil
+						}
+					}
+
 					select {
 					case chunks <- swarm.NewChunk(swarm.NewAddress(dataItem.Address), dataItem.Data).WithTagID(item.Tag):
 						count++
-						// set next iteration start item
-						// when its chunk is successfully sent to channel
-						sinceItem = &item
+						// when the chunk is successfully sent to channel
```

Comment:
this comment is not really necessary, the code is self explanatory

Reply:
comment left from previous code, will remove it